### ISSUE TITLE
Resource pack cache is broken when reading datapacks

### ISSUE THREAD

**Issue by RaKGEKKO** [SettingDust/lazyyyyy#11]
[crash-2024-12-29_06.25.07-client.txt](https://github.com/user-attachments/files/18270756/crash-2024-12-29_06.25.07-client.txt)

When i start the game nothing weird happens, but when i try to create a world it crashes.

**Comment by SettingDust** [comment]
show me the log

**Comment by RaKGEKKO** [comment]
uhh its there, its the file called crash-2024-12-29-06.25.07-client.txt

**Comment by SettingDust** [comment]
Log is files in `logs`. Crash report isn't log. I can reproduce when join world. I'll look into it

**Comment by RaKGEKKO** [comment]
[latest.log](https://github.com/user-attachments/files/18284052/latest.log)
here you go

**Comment by SettingDust** [comment]
Should be fixed in 0.5.10. Give it a test

**Comment by RaKGEKKO** [comment]
Okay, ill try it when its out.

**Comment by RaKGEKKO** [comment]
It seems to be fixed, thanks.

have a nice day.

**Comment by SettingDust** [comment]
For curious. Can you show me the improvement if there is?

**Comment by RaKGEKKO** [comment]
I sadly can't calculate that but with or without the mod it variates between 23 and 30 seconds. but i don't think that's accurate for your mod because i am using a really large modpack so you should do what other performance mods do, calculate launching speed with and without some performance mods like sodium so you put the statistics on your main page.

**Comment by SettingDust** [comment]
That's weird. It's reduced 150-200 secs for my 700 mods pack. I'm optimizing the point that the others haven't touched

**Comment by RaKGEKKO** [comment]
Well i have a very weird modpack, its really frustrating because it works well but has some bugs that break it, like when i use resourcepacks it doesn't fully load so i gotta reset the game for it to apply. I would like to fix it but i have low to none knowledge about this things, but it is really optimized 60 fps almost every time and 255 max fps.

**Comment by RaKGEKKO** [comment]
I don't really know, BTW I was testing some things and it worked fine, but the CPU usage when i am generating forests/mountains, go up to 80%+ after installing lazyyyyy, maybe its an error? Or is it normal?

**Comment by SettingDust** [comment]
I don't think it's a bug relating to lazyyyyy. Maybe you need to try to reproduce without lazyyyyy with the same seed

**Comment by RaKGEKKO** [comment]
Yeah maybe it doesn't need anything to do with lazyyyyy, anyways great mod.

**Comment by RaKGEKKO** [comment]
Hope to fix all my bugs some day.
